### PR TITLE
Unify error messages for omitted type arguments in extension method calls

### LINKED ISSUE #13617: Invalid generic extension call does not report error and then fail to emit module.
**Version Used**: 

VS2015 Update 3 with .NET Core Tools Preview 2

**Steps to Reproduce**:
1. Create a generic extension method which return `T` e.g. `public static T Foo<T>(this object obj) => default(T)`
2. Call it somewhere without specifying `T` and invoke any arbitrary member name, like `"abc".Foo<>().FooValue`
3. Compile.

**Expected Behavior**:

Compilation fail with proper error message.

**Actual Behavior**:

No typical compilation error reported but fail to build with error CS7038 : Fail to emit module

**Sample class**:

```
public static class FooExtensions
{
    public static T Foo<T>(this object obj) => default(T);
}

public class Class1
{
    public void Test()
    {
        var a = "a".Foo<>().Run();
        // the line below fail to compile as expected
        //var b = FooExtensions.Foo<>().Abc;
    }
}
```

https://dotnetfiddle.net/JvacT3

### PR DESCRIPTION
**Pull request by OmarTawfik**
This PR reverts #15718, and proposes a better way to fix #13617 (as suggested by @AlekseyTs).

### CODE REVIEW

**Comment by OmarTawfik** [comment]
@dotnet/roslyn-compiler @AlekseyTs PR ready for review.

File: src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs [hunk 1]
```diff
@@ -5697,9 +5690,25 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
...
+                        nodeHasOmittedTypeArgs = true;
+                    }
+                }
+            }
+
             int arity;
             LookupOptions options;
-            if (typeArguments.IsDefault)
+
+            if (typeArguments.IsDefault || nodeHasOmittedTypeArgs)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs]: > || nodeHasOmittedTypeArgs [](start = 39, length = 26)

This feels suspicious. I think lookup should still respect requested arity even when there are omitted arguments.

File: src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs [hunk 2]
```diff
@@ -119,18 +119,18 @@ void M()
     }
 }";
             CreateCompilationWithMscorlib(source).VerifyDiagnostics(
-                // (9,9): error CS0305: Using the generic method 'C.M1<T>()' requires 1 type arguments
-                Diagnostic(ErrorCode.ERR_BadArity, "M1<object, object>").WithArguments("C.M1<T>()", "method", "1").WithLocation(9, 9),
-                // (10,11): error CS0305: Using the generic method 'C.M1<T>()' requires 1 type arguments
-                Diagnostic(ErrorCode.ERR_BadArity, "M1<object, object>").WithArguments("C.M1<T>()", "method", "1").WithLocation(10, 11),
+                // (9,9): error CS0305: Using the generic method group 'C.M1<T>()' requires 1 type arguments
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs]: >// (9,9): error CS0305: Using the generic method group 'C.M1<T>()' requires 1 type arguments [](start = 16, length = 92)

Why do we add "group" here?

**Comment by AlekseyTs** [comment]
@OmarTawfik Let's discuss this change tomorrow.

**Comment by OmarTawfik** [comment]
After discussion offline, we're OK with keeping the old error messages. Will send another PR shortly.

**Pull request closed without merging**